practices/appeng/adrs/0002-distributed-tracing.md

```diff
@@ -0,0 +1,156 @@
+# 0002 - Use OpenTelemetry to instrument code for distributed tracing
+
+**Status:** In Progress
+
+**Date Superseded:** (if applicable)
+
+**Date Accepted:**
+
+**Reviewers:** @jim @reggieriser @rogeruiz \<tag yourself!>
+
+## Context
+
+### Objective
+
+_Give developers and operators an easier way to understand the behavior and
+structure of running systems by instrumenting code for distributed tracing._
+
+### Why Distributed Tracing?
+
+> Observability is defined as the ability of the internal states of a system to
+> be determined by its external outputs.
+
+There are several established techniques for gaining observability into complex
+software systems.
+
+- Logging
+  - helpful for known knowns
+  - e.g. "I want to know when a specific condition is reached"
+- Metrics
+  - helpful for assessing known unknowns
+  - e.g. "How many requests per second was the system handling last Tuesday?"
+- Distributed Tracing
+  - helpful for learning about unknown unknowns
+  - e.g. "What was the execution context for User X that caused their
+    interaction to timeout last Tuesday?"
+
+Some of the benefits of distributed tracing, as outlined in
+[this](https://petabridge.com/blog/why-use-distributed-tracing/) article are:
```

Comment:
Small nitpick but it's generally good to include more descriptive text in links for accessibility. See this [A11y Project post on accessible links](https://www.a11yproject.com/posts/2019-02-15-creating-valid-and-accessible-links/).

Reply:
Good point.  I changed this to reference the title of the article.